src/main/java/edu/wpi/grip/core/events/NotificationEvent.java

```diff
@@ -0,0 +1,31 @@
+package edu.wpi.grip.core.events;
+
+import java.util.Optional;
+
+/**
+ * Should be posted when a non fatal user cause error occurs.
+ * This is used to communicate to the main window that a
+ * error should be displayed unobtrusively to the user.
+ * The message that is displayed is the message in the Throwable.
+ */
+public class NotificationEvent {
+    private final Throwable exception;
+    private final Optional<String> detailedDescription;
+
+    public NotificationEvent(Throwable exception){
+        this(exception, null);
+    }
+
+    public NotificationEvent(Throwable exception, String detailedDescription) {
+        this.exception = exception;
```

Comment:
Use `checkNotNull(exception, "The exception can not be null");` from googles api